### ISSUE TITLE
ZFS pool not mounted on boot on Ubuntu 14.04.1 (trusty)

### ISSUE THREAD

**Issue by Zsub** [zfsonlinux/zfs#2556]
After buying a new PC, my exported and imported ZFS pool doesn't mount on boot anymore using Ubuntu 14.04.1. I have an SSD I use for `/` and for the ZIL and L2arc, and three WD Greens in RAIDZ1.

I installed using the stable ppa, through `apt-get install ubuntu-zfs`. I have also double-checked all the steps in the [mountall-howto](https://github.com/zfsonlinux/pkg-zfs/wiki/Ubuntu-ZFS-mountall-FAQ-and-troubleshooting), but to no avail. It certainly does not 'just work' :-)

I have attached all the output as described in the final step of the mountall howto in this [gist](https://gist.github.com/Zsub/eb018ba4e953fc85650f).

The only thing seems to be that the filesystem just doesn't get mounted, zfs knows about it and everything.

Thanks for taking a look!

**Comment by borutmrak** [comment]
I have this same issue on multiple (not all) machines running Ubuntu 14.04.

One (my personal machine) has a SSD rpool (single drive) and raidz1 data pool (2x 1TB drives) - all of those are vdevs on LUKS. rpool mounts, data pool has datasets that are peppered all over (/home/username/Downloads etc). rpool datasets all mount, but none from the other pool.

Another machine has ext4 root on SSD and a similar (2x 1TB) raidz1 pool that has to be mounted manually after boot.

I was thinking that it might have something to do with the mount order. For instance on the first machine "zfs list" first lists the data pool and after that the rpool. If it tries to mount in the order listed, the mounpoints do not exist yet. I *think* I solved this on one other machine by renaming the datasets to change the sort list and it seems to work now.

The second machine, however, has all the mountpoints there (when ext4 root is mounted) but still requires manual "zfs mount -a" after boot.

This is zfs list from the first machine (I removed a few lines that I don't think are important here):

➜  ~  sudo zfs list
NAME                                   USED  AVAIL  REFER  MOUNTPOINT
data                                   526G   389G    39K  /data
data/home                              443G   389G  10.4G  none
data/home/b                            432G   389G  6.42G  legacy
data/home/b/Downloads                  140G   389G   140G  /home/b/Downloads
[snip]
data/home/b/cold-storage              18.9G   389G  18.9G  /home/b/cold-storage
rpool                                 58.4G  39.5G    33K  /rpool
rpool/ROOT                            17.2G  39.5G    32K  /rpool/ROOT
rpool/ROOT/ubuntu-trusty              17.2G  39.5G  10.5G  /
rpool/home                            20.4G  39.5G    30K  /rpool/home
rpool/home/b                          20.4G  39.5G  15.2G  /home/b
rpool/vmimages                        20.6G  39.5G    30K  /rpool/vmimages
rpool/vmimages/vm-win7                20.6G  40.0G  20.2G  -


I can reproduce this on every boot, I'll be happy to test anything you throw my way.

Using initramfs-tools though, not dracut (that one is not really useful as packaged in Ubuntu).

thanks,
B.
